src/main/java/com/google/api/gax/bundling/ThresholdBundler.java

```diff
@@ -179,7 +195,13 @@ public int drainNextBundleTo(Collection<? super E> outputCollection) {
       }
 
       if (outBundle != null) {
-        outputCollection.addAll(outBundle.getData());
+        List<E> data = outBundle.getData();
+        outputCollection.addAll(data);
+        if (flowController != null) {
```

Comment:
I think this if-block should be moved above `addAll`. This way flow controller still works properly even if `addAll` somehow throws an exception.